biom/parse.py

```diff
@@ -245,6 +246,98 @@ def get_axis_indices(biom_str, to_keep, axis):
     return idxs, json.dumps(subset)[1:-1]  # trim off { and }
 
 
+def parse_uc(fh):
+    """ Create a Table object from a uclust/usearch/vsearch uc file.
+
+        Parameters
+        ----------
+        fh : file handle
+            The ``.uc`` file to be parsed.
+
+        Returns
+        -------
+        biom.Table : The resulting BIOM table.
+
+        Raises
+        ------
+        ValueError
+            If a sequence identifier is encountered that doesn't have at least
+            one underscore in it (see Notes).
+
+        Notes
+        -----
+        This file assumes that sequence identifiers in this file are in QIIME's
+        "post-split-libraries" format, where the identifiers are of the form
+        ``<sample-id>_<sequence-id>``. Everything before the first underscore
+        will be used as the sample identifier in the resulting ``Table``.
+        The information after the first underscore is not used directly, though
+        the full identifiers of seeds will be used as the observation
+        identifier in the resulting ``Table``.
+
+    """
+    data = collections.defaultdict(int)
+    sample_idxs = {}
+    sample_ids = []
+    observation_idxs = {}
+    observation_ids = []
+    # The types of hit lines we need here are hit (H), seed (S) and
+    # library seed (L). Store these in a set for quick reference.
+    line_types = set(list('HSL'))
+    for line in fh:
+        # determine if the current line is one that we need
+        line = line.strip()
+        if not (line and line[0] in line_types):
+            # if the line is blank, or not one of the types we need,
+            # including comments, move on to the next line
+            continue
+        else:
+            fields = line.split('\t')
+
+        # grab the fields we care about
+        hit_type = fields[0]
+        observation_id = fields[9].split()[0]
+        query_id = fields[8].split()[0]
+
+        if observation_id == '*':
+            # S and L lines don't have a separate observation id
+            observation_id = query_id
+
+        # get the index of the current observation id, or create it if it's
+        # the first time we're seeing this id
+        try:
+            observation_idx = observation_idxs[observation_id]
```

Comment:
Since this is expected to happen frequently, could this be done as a test for membership?